src/js/background.js

```diff
@@ -239,9 +239,15 @@ Badger.prototype = {
       };
     }
 
+    let host = window.extractHostFromURL(frameUrl);
+    //if frame URL is 'about:blank', inherit parent host
+    if (frameUrl === "about:blank") {
+      host = self.tabData[tabId].frames[parentFrameId].host;
```

Comment:
Would we always have frame data for the parent frame?

We build `tabData` inside `chrome.webRequest.onBeforeRequest`. We wouldn't have received [`onBeforeRequest` events for inline HTML frames](https://bugs.chromium.org/p/chromium/issues/detail?id=838242) nor for cached resources (they bypass `chrome.webRequest`). Any other cases? Unfortunately, `tabData.frames` doesn't contain complete frame data at this time.

So if we have a page with a third-party domain frame that itself has a nested inline HTML frame that creates an `about:blank` frame that uses localStorage, we'll get a `parentFrameId` here that we don't have indexed in `tabData.frames`.

I haven't verified this though.

Reply:
Also, what if this is an "about:blank" frame nested inside one or more "about:blank" frames?

Reply:
Firefox gives us [`frameAncestors`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onBeforeRequest#details) as of Firefox 58. It might help, but it's not a cross-browser solution.

Reply:
It would be great if we could make `tabData.frames` reliable but if we can't for now, should we instead travel up the document tree inside the localStorage content script and get the first http/https parent frame's URL and send that together with what we currently send to the background page?

Reply:
Well darn, I thought this was an elegant solution but not consistently having a complete frame hierarchy makes it not work.  (I was assuming that we always had a complete hierarchy, and that it was always built from the top down, which would have meant that even nested `about:blank` frames would get a valid host, but alas...)

Reply:
Yep, same, I thought this approach is smooth ... except `tabData` is incomplete.

Reply:
From the bit of testing that I've done, it seems that supercookie.js is NOT injected into at least some `about:blank` iframes (it may depend on the frame contents and how they are loaded).  Any localStorage set in the iframe "leaks" to the containing frame as well.  Not sure if this info is useful, but here it is for posterity.

Reply:
This may be related to [`match_about_blank`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts), which we don't use at this time (but probably should). But since we have this problem now, `match_about_blank` would be a separate discussion.

Reply:
By the way, I haven't confirmed my analysis, this is just from reading the code. I could be wrong and this approach can be made to work after some adjustments. This is why setting up a few controlled test cases would be so useful.